### PR TITLE
Add provider equinix and module_name in Equinix Metal TF configs

### PR DESCRIPTION
**Pull request by codinja1188**

### CODE REVIEW

File: examples/fabric-port-connection-with-zside-token/versions.tf [hunk 1]
```diff
@@ -4,7 +4,10 @@ terraform {
   required_providers {
     equinix = {
       source  = "equinix/equinix"
-      version = ">= 1.7.0"
+      version = "~> 1.14"
     }
   }
+  provider_meta "equinix" {
+    module_name = "equinix-fabric-connection-metal/fabric-port-connection-with-zside-token"
+  }
```
Comment (ocobleseqx) [inline examples/fabric-port-connection-with-zside-token/versions.tf:12]: Examples don't need to define provider_meta, it's just for the module itself
```suggestion
```

File: examples/fabric-port-connection/versions.tf [hunk 2]
```diff
@@ -4,7 +4,10 @@ terraform {
   required_providers {
     equinix = {
       source  = "equinix/equinix"
-      version = ">= 1.7.0"
+      version = "~> 1.14"
     }
   }
+  provider_meta "equinix" {
+    module_name = "equinix-fabric-connection-metal/fabric-port-connection"
+  }
```
Comment (ocobleseqx) [inline examples/fabric-port-connection/versions.tf:12]: Examples don't need to define provider_meta, it's just for the module itself
```suggestion
```

File: examples/network-edge-device-redundant-connection/versions.tf [hunk 3]
```diff
@@ -4,7 +4,10 @@ terraform {
   required_providers {
     equinix = {
       source  = "equinix/equinix"
-      version = ">= 1.7.0"
+      version = "~> 1.14"
     }
   }
+  provider_meta "equinix" {
+    module_name = "equinix-fabric-connection-metal/network-edge-device-redundant-connection"
+  }
```
Comment (ocobleseqx) [inline examples/network-edge-device-redundant-connection/versions.tf:12]:
```suggestion
```

File: examples/service-token-redundant-connection/versions.tf [hunk 4]
```diff
@@ -4,7 +4,10 @@ terraform {
   required_providers {
     equinix = {
       source  = "equinix/equinix"
-      version = ">= 1.7.0"
+      version = "~> 1.14"
     }
   }
+  provider_meta "equinix" {
+    module_name = "equinix-fabric-connection-metal/service-token-redundant-connection"
+  }
```
Comment (ocobleseqx) [inline examples/service-token-redundant-connection/versions.tf:12]:
```suggestion
```

**Review by ocobleseqx — DISMISSED** [review]
Please, review requested changes

**Comment by vasubabu** [comment]
> Please, review requested changes

updated

**Review by ctreatma — APPROVED** [review]

**Comment by ctreatma** [comment]
Approved but my approval doesn't count for anything on this repo 😢

**Review by ocobleseqx — APPROVED** [review]
👍🏼

**Pull request merged**